### PR TITLE
9 prepare for production deployment

### LINKED ISSUE #9: Prepare for production deployment
Docker to hold containers for each app.
Nginx for the client app.
Possibly Gunicorn for the web server. TBD

### PR DESCRIPTION
**Pull request by LukeParky**
This PR Allows the app to run on a production server. The docker related files should be easy to read and so should any documentation.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,7 +6,7 @@ Geospatial Research Institute Toi Hanagarau. This application is used to visuali
 ![image](https://user-images.githubusercontent.com/41398636/126084844-1bad1906-8784-4f9b-93ec-df0e5e995dd6.png)
```
Comment (rosepearson) [inline README.md:8]: I think this is getting close to where you might want to split it out into your wiki (or other documentation). I wonder about having two sections in your wiki - package usage (the first 4/5 of your README.md), and package developers (the contents already in the wiki + the contributions, license section of your README.md)

**Review by rosepearson — COMMENTED** [review]
Hi @LukeParky I've had a quick read through all your file changes. They are clear and generally look good to me. I'll pull down the repo and have a look at the files and changes in more detail + do a bit of reading about Docker as it's ears since i've done anything with it and then it was very brief. I've chatted to @Pooja3894 and she will test the depolyment based on your README instructions. 

One thing I think you should add is a pull request template - this is a marddown file with template contents for a PR. I think this could contain some of the material that will eventually also exist int he wiki/PR sections. You could have a 'developer checklist' section, and a 'reviewer checklist' section. You can just delete any of the template sections that aren't relevant to a particular PR. You can see [example template](https://github.com/niwa/geoapis/blob/main/.github/pull_request_template.md) for an example - inspired by one of your reviews of my code :)

**Comment by LukeParky** [comment]
> One thing I think you should add is a pull request template

Thanks Rose, I'll add this now

File: docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+        build: ./server
+        ports:
+            - "5000:5000"
+    www:
+        build:
+            context: ./client
+            args:
+                HERE_API: ${HERE_MAPS_TOKEN}
+        ports:
+            - "80:80"
```
Comment (Pooja3894) [inline docker-compose.yml:13]: port 80 wasn't working on my machine, So changed it to 8080:80

File: server/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,8 @@
+# syntax=docker/dockerfile:1
+FROM python:3-slim
+WORKDIR /app
+COPY requirements.txt ./
+RUN pip3 install --no-cache-dir -r requirements.txt
+COPY ./ .
+EXPOSE 5000
```
Comment (Pooja3894) [inline server/Dockerfile:7]: EXPOSE is useful if we are using multi-containers application. One benefit of using the EXPOSE instruction, whether you are running a multi-container application or not, is that it helps others understand what port the application listens by just reading the Dockerfile without the need of going through your code.

**Review by Pooja3894 — APPROVED** [review]
Docker is up and running successfully. Great work @LukeParky

File: docker-compose.yml (hunk 2 above)
Comment (LukeParky) [reply docker-compose.yml:13]: In the instructions in the readme I mentioned ensuring port 80 is open. Do you think this is enough or should I add something more?

File: server/Dockerfile (hunk 3 above)
Comment (LukeParky) [reply server/Dockerfile:7]: I wasn't 100% sure if it was necessary but I liked that it makes people understand what ports it's running on easier

File: README.md (hunk 1 above)
Comment (LukeParky) [reply README.md:8]: I agree, I've added a documentation issue #21 to make sure I get around to this before working on the next part of the project

Comment (rosepearson) [reply README.md:8]: Can be done in a separate issue.

**Pull request merged**